File: backend/gradle/src/main/java/com/dreamteam/atelier/domain/UserAddress.java
package com.dreamteam.atelier.domain;

import java.util.UUID;

/**
 * Базовый тип описывающий сущность - адрес пользователя
 */
public class UserAddress {

    private String id;
    private UUID uuid;
    private String region;
    private String city;
    private String street;
    private String house;
    private String flat;
    private String description; //дополнительное описание

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public UUID getUuid() {
        return uuid;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getHouse() {
        return house;
    }

    public void setHouse(String house) {
        this.house = house;
    }

    public String getFlat() {
        return flat;
    }

    public void setFlat(String flat) {
        this.flat = flat;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: backend/nodejs/src/graphql/types/user-address.ts
import { GraphQLID, GraphQLObjectType, GraphQLString } from 'graphql';

export const UserAddressType = new GraphQLObjectType({
    name: 'userAddress',
    fields: {
        id: {type: GraphQLID},
        region: {type: GraphQLString},
        city: {type: GraphQLString},
        street: {type: GraphQLString},
        house: {type: GraphQLString},
        floor: {type: GraphQLString},
        flat: {type: GraphQLString},
        description: {type: GraphQLString}
    }
});
